File: cdm/csharp/properties/SEScalar0To1.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

public class SEScalar0To1 : SEScalar
{
  public SEScalar0To1() : base()
  {

  }

  public SEScalar0To1(SEScalar0To1 from)
  {
    Set(from);
  }

  public SEScalar0To1(double value)
  {
    SetValue(value);
  }

  public void Set(SEScalar0To1 from)
  {
    SetValue(from.value);
  }

  public override void SetValue(double val)
  {
    if (val >= 0 && val <= 1)
      value = val;
    else
      throw new System.Exception("Value must be [0-1]");
  }
}

File: src/cpp/cdm/properties/SEScalar0To1.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "properties/SEScalar0To1.h"

SEScalar0To1::SEScalar0To1() : SEScalar()
{

}

void SEScalar0To1::SetValue(double d)
{ 
  if (d > 1 || d < 0)
    throw CommonDataModelException("SEScalar0To1 must be between [0,1]");
  SEScalar::SetValue(d);
}

void SEScalar0To1::SetValue(double d, const NoUnit& unitless)
{ 
  SEScalar0To1::SetValue(d);
}

void SEScalar0To1::ForceValue(double d)
{
  SEScalar::ForceValue(d);
}

void SEScalar0To1::ForceValue(double d, const NoUnit& unitless)
{
  SEScalar::ForceValue(d);
}
